apps/status-api/src/controllers/OverallStatusController.ts

```diff
@@ -0,0 +1,47 @@
+import { Controller, Get } from '@nestjs/common'
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { BlockchainStatus, BlockchainStatusController } from './BlockchainStatusController'
+import { WhaleApiProbeIndicator } from '../modules/WhaleApiModule'
+
+@Controller('overall')
+export class OverallStatusController {
+  constructor (
+    private readonly client: WhaleApiClient,
+    private readonly probe: WhaleApiProbeIndicator,
+    private readonly blockchainStatusController: BlockchainStatusController
+  ) {
+  }
+
+  /**
+   *  To provide overall status for Ocean and blockchain services. Returns
+   *  'operational' when both services are up,
+   *  'outage' when either service is down and
+   *  'degraded' when blockchain service is degraded.
+   *
+   *  @return {Promise<{ status: BlockchainStatus }>}
+   */
+
+  @Get()
+  async getOverallStatus (): Promise<{ status: BlockchainStatus }> {
+    const liveness = await this.probe.liveness()
+
+    if (liveness.whale.status === 'down') {
+      return {
+        status: 'outage'
+      }
+    }
+
+    const readiness = await this.probe.readiness()
+
+    if (readiness.whale.status === 'down') {
+      return {
+        status: 'outage'
+      }
+    }
```

Comment:
Just some refactoring
```suggestion
    const liveness = await this.probe.liveness()
    const readiness = await this.probe.readiness()
    
    if (liveness.whale.status === 'down' || readiness.whale.status === 'down') {
      return {
        status: 'outage'
      }
    }
```

Reply:
Maybe it is better not to use the same status

Reply:
boolean short circuit can prevent the second redundant call
```
if (
  (await this.probe.liveness()).whale.status === 'down' || 
  (await this.probe.readiness()).whale.status === 'down'
) {
  return {
    status: 'outage'
  }
}
```

Albeit at the expense of some readability